src/jit/importer.cpp

```diff
@@ -1480,10 +1480,16 @@ var_types Compiler::impNormStructType(CORINFO_CLASS_HANDLE structHnd,
     assert(structHnd != NO_CLASS_HANDLE);
 
     const DWORD structFlags = info.compCompHnd->getClassAttribs(structHnd);
-    const bool  isRefAny    = (structHnd == impGetRefAnyClass());
-    const bool  hasGCPtrs   = isRefAny || ((structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0);
     var_types   structType  = TYP_STRUCT;
 
+#ifndef FEATURE_CORECLR
+    // Desktop CLR won't report FLG_CONTAINS_GC_PTR for RefAnyClass - need to check explicitly.
+    const bool isRefAny  = (structHnd == impGetRefAnyClass());
+    const bool hasGCPtrs = isRefAny || ((structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0);
+#else
+    const bool        hasGCPtrs                                        = (structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0;
+#endif
```

Comment:
For clarity/readability of #ifdef condition, i would suggest to change it as follows:

#ifdef FEATURE_CORECLR
    const bool        hasGCPtrs                                        = (structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0;
#else
    // Desktop CLR won't report FLG_CONTAINS_GC_PTR for RefAnyClass - need to check explicitly.
    const bool isRefAny  = (structHnd == impGetRefAnyClass());
    const bool hasGCPtrs = isRefAny || ((structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0);
#endif